src/site/stages/build/plugins/create-environment-filter.js

```diff
@@ -1,11 +1,17 @@
 /* eslint-disable no-param-reassign */
+const ENVIRONMENTS = require('../../../constants/environments');
 
 function createEnvironmentFilter(options) {
   const environmentName = options.buildtype;
 
   return (files, metalsmith, done) => {
     for (const fileName of Object.keys(files)) {
       const file = files[fileName];
+
+      if (environmentName !== ENVIRONMENTS.LOCALHOST) {
+        if (file.draft) delete files[fileName];
+      }
+
```

Comment:
There's more context in the ticket, but this `draft` property makes it safe for content editors to merge unfinished pages into `master` for Drupal to pick up.